source/views/sis/course-search/search.js

```diff
@@ -76,7 +125,20 @@ class CourseSearchView extends React.PureComponent<Props, State> {
 		applyFilters: applyFiltersToItem,
 	}
 
+	static getDerivedStateFromProps(nextProps: Props, prevState: State) {
+		if (prevState.query && nextProps.filters !== prevState.cachedFilters) {
+			return executeSearch({
+				text: prevState.query,
+				filters: nextProps.filters,
+				applyFilters: nextProps.applyFilters,
+				allCourses: nextProps.allCourses,
+				updateRecentSearches: nextProps.updateRecentSearches,
+			})
+		}
```

Comment:
Add "return null" after this line, so the method returns null if the condition doesn't pass.